import os, glob, sys
from snakemake.utils import R

SAMPLEMAPPING = 'ICUsamples.tsv'
# SAMPLEMAPPING = 'sample_mapping.tsv'
FASTQDIR = 'sra/'
OUTDIR   = 'ICU-out1/'
TMPDIR   = 'tmp/'

ORGANISM = 'H_sapiens_hg19'

CLIPTRIMOUT = FASTQDIR
REMOVEPCRDUBLICATESOUT = OUTDIR + 'removed_pcr_dub/'
MPILEUPIN = REMOVEPCRDUBLICATESOUT

include: '../../snake/wes/wes_snake.py'

SAMPLENAMES = getSampleNames()

rule SRA051153:
    input:
       expand(HAPLOTYPECALLEROUT + '{sample}.g.vcf', sample = SAMPLENAMES),
       expand(HAPLOTYPECALLEROUT + '{sample}.vcf', sample = SAMPLENAMES),
       expand(HAPLOTYPECALLEROUT + '{sample}_snps.vcf', sample = SAMPLENAMES),
       expand(HAPLOTYPECALLEROUT + '{sample}_indels.vcf', sample = SAMPLENAMES),
       expand(HAPLOTYPECALLEROUT + '{sample}_filtered_snps.vcf', sample = SAMPLENAMES),
       expand(HAPLOTYPECALLEROUT + '{sample}_filtered_indels.vcf', sample = SAMPLENAMES),
       expand(ANNOVAROUT+ '{sample}.hg19_multianno.vcf', sample= SAMPLENAMES) 
